google-cloud-clients/google-cloud-contrib/google-cloud-spanner-jdbc/src/main/java/com/google/cloud/spanner/jdbc/AbstractJdbcPreparedStatement.java

```diff
@@ -0,0 +1,388 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.spanner.jdbc;
+
+import java.io.InputStream;
+import java.io.Reader;
+import java.math.BigDecimal;
+import java.net.URL;
+import java.sql.Array;
+import java.sql.Blob;
+import java.sql.Clob;
+import java.sql.Date;
+import java.sql.NClob;
+import java.sql.PreparedStatement;
+import java.sql.Ref;
+import java.sql.ResultSet;
+import java.sql.ResultSetMetaData;
+import java.sql.RowId;
+import java.sql.SQLException;
+import java.sql.SQLXML;
+import java.sql.Time;
+import java.sql.Timestamp;
+import java.sql.Types;
+import java.util.Calendar;
+import com.google.rpc.Code;
+
+/** Base class for Cloud Spanner {@link PreparedStatement}s. */
+abstract class AbstractJdbcPreparedStatement extends JdbcStatement implements PreparedStatement {
+  private static final String METHOD_NOT_ON_PREPARED_STATEMENT =
+      "This method may not be called on a PreparedStatement";
+  private final JdbcParameterStore parameters = new JdbcParameterStore();
+
+  AbstractJdbcPreparedStatement(JdbcConnection connection) {
+    super(connection);
+  }
+
+  JdbcParameterStore getParameters() {
+    return parameters;
+  }
+
+  private <T> T checkClosedAndThrowNotOnPreparedStatement() throws SQLException {
+    checkClosed();
+    throw JdbcSqlExceptionFactory.of(METHOD_NOT_ON_PREPARED_STATEMENT, Code.INVALID_ARGUMENT);
+  }
+
+  @Override
+  public ResultSet executeQuery(String sql) throws SQLException {
+    return checkClosedAndThrowNotOnPreparedStatement();
+  }
+
+  @Override
+  public int executeUpdate(String sql) throws SQLException {
+    return checkClosedAndThrowNotOnPreparedStatement();
+  }
+
+  @Override
+  public boolean execute(String sql) throws SQLException {
+    return checkClosedAndThrowNotOnPreparedStatement();
+  }
+
+  @Override
+  public void addBatch(String sql) throws SQLException {
+    checkClosedAndThrowNotOnPreparedStatement();
+  }
+
+  @Override
+  public void setNull(int parameterIndex, int sqlType) throws SQLException {
+    checkClosed();
+    parameters.setParameter(parameterIndex, null, sqlType, null);
+  }
+
+  @Override
+  public void setBoolean(int parameterIndex, boolean x) throws SQLException {
```

Comment:
minor - please rename all `x` variables - `value` maybe?

Reply:
`x` variables have been renamed to `value`.